tests/test_events.py

```diff
@@ -224,16 +227,16 @@ def test_process_block_range(client, base_ddo_url, events_object):
         assert events_object.process_current_blocks() is None
 
 
-def test_get_last_processed_block(events_object):
-    start_block = events_object._start_block
+def test_get_last_processed_block(monkeypatch, events_object):
+    monkeypatch.setenv("REDIS_CONNECTION", "redis://172.15.0.18:6379")
```

Comment:
you need to test more cases. One of the cases I suggested was the one where ES works properly, then fails only at the n-th call. You can see examples of how to do that in the provider algo validation tests.